node/http_test.ts

```diff
@@ -148,3 +148,150 @@ Deno.test("[node/http] request default protocol", async () => {
   });
   await promise;
 });
+
+Deno.test("[node/http] send request with non-chunked body", async () => {
+  const promise = deferred<void>();
+  let requestHeaders: Record<string, string> = {};
+  const requestBody: string[] = [];
+
+  const server = http.createServer((req, res) => {
+    req.on("data", (chunk) => {
+      requestBody.push(chunk.toString());
+    });
+    req.on("end", () => {
+      requestHeaders = Object.fromEntries(
+        Object.entries(req.headers).map(([k, v]) => [k.toLowerCase(), v]),
+      );
+      res.end("ok");
+    });
+  });
+  server.listen(() => {
+    const opts: RequestOptions = {
+      host: "localhost",
+      port: server.address().port,
+      method: "POST",
+      headers: {
+        "Content-Type": "text/plain; charset=utf-8",
+        "Content-Length": "11",
+      },
+    };
+    const req = http.request(opts, (res) => {
+      res.on("data", () => {});
+      res.on("end", () => {
+        server.close();
+      });
+      assertEquals(res.statusCode, 200);
+      assertEquals(requestHeaders["content-length"], "11");
+      assertEquals(
+        /(?:^|\W)chunked(?:$|\W)/i.test(requestHeaders["transfer-encoding"]),
+        false,
+      );
+      assertEquals(requestBody[0], "hello world");
+    });
+    req.write("hello ");
+    req.write("world");
+    req.end();
+  });
+  server.on("close", () => {
+    promise.resolve();
+  });
+  await promise;
+});
+
+Deno.test("[node/http] send request with chunked body", async () => {
```

Comment:
I am aware of this.
I think it is difficult to make the behavior completely the same as Node because of the spec of Deno's fetch().

__For Node's http.request():__
If both `Content-Length` and `Transfer-Encoding` headers are specified, both `Content-Length` and `Transfer-Encoding` headers are sent. The request body is chunked encoding.

__For Deno's fetch():__
Even if both `Content-Length` and `Transfer-Encoding` headers are specified(*), if the body is a ReadableStream, only the `Transfer-Encoding` header is sent.

This unit test is designed to make sure that the `Content-Length` header is not sent depending on the current behavior of Deno's fetch(), so it is not passed by Node.

For reference, the HTTP logs obtained by Wireshark during the test run are attached below.

_Node:_

```
POST / HTTP/1.1
Content-Type: text/plain; charset=utf-8
Content-Length: 11
Transfer-Encoding: chunked
Host: localhost:56750
Connection: close

6
hello 
5
world
0

HTTP/1.1 400 Bad Request
Connection: close

```

_Deno:_

```
POST / HTTP/1.1
content-type: text/plain; charset=utf-8
transfer-encoding: chunked
accept: */*
accept-language: *
user-agent: Deno/1.26.1
accept-encoding: gzip, br
host: localhost:56390

6
hello 
5
world
0

HTTP/1.1 200 OK
Date: Fri, 07 Oct 2022 19:28:48 GMT
Content-Type: text/plain;charset=UTF-8
Content-Length: 2

ok
```

(*) The current implementation does not pass the headers option to fetch() in the relevant code of node/http.ts, but the same result will be obtained if it is changed to do so.
https://github.com/denoland/deno_std/pull/2755/files#diff-07717ecd86915a087f8759f720b0217ab7c3c9f473aa8f4772aeae5dbb638a92R127-R131
I would like to create a separate PR for not passing the headers option to fetch().

Reply:
@PolarETech Do we need this 2nd test case? The first and third ones look enough for checking the fix

Reply:
Ok, I confirmed that Node.js sends `transfer-encoding: chunked` when both `content-length: N` and `transfer-encoding: chunked` are specified. This test case seems useful to me. But how about rewriting the server part of the test case with `serve` function of `std/http`? Because this test case depends on non-compliant behavior of `http.Server` and server part is not the subject of the test case, I think it makes more sense to use `std/http` here.

Reply:
Thanks for your advice. I have changed the unit test cases. (d38c018)